include/clue/shared_mutex.hpp

```diff
@@ -356,24 +358,26 @@ class shared_lock {
 
     void lock() {
         mut_->lock_shared();
+        owns_ = true;
     }
 
-    void try_lock() {
-        mut_->try_lock_shared();
+    bool try_lock() {
+        return owns_ = mut_->try_lock_shared();
```

Comment:
For code style purpose, don't write ``return a = b``, instead please rewrite this as:
```c++
owns_ = mut_->try_lock_shared();
return owns_;
```